plugins/playstation/src/playstation-plugin.vala

```diff
@@ -0,0 +1,36 @@
+// This file is part of GNOME Games. License: GPLv3
+
+private class Games.PlayStation : Object, Plugin {
+	private const string FINGERPRINT_PREFIX = "playstation";
+	private const string MIME_TYPE = "application/x-cue";
+	private const string MODULE_BASENAME = "libretro-playstation.so";
+	private const bool SUPPORTS_SNAPSHOTTING = true;
+
+	public GameSource get_game_source () throws Error {
+		var query = new MimeTypeTrackerQuery (MIME_TYPE, game_for_uri);
+		var connection = Tracker.Sparql.Connection.@get ();
+		var source = new TrackerGameSource (connection);
+		source.add_query (query);
+
+		return source;
+	}
+
+	private static Game game_for_uri (string uri) throws Error {
+		var file = File.new_for_uri (uri);
+		var validator = new PlayStationHeader (file);
+		validator.check_validity ();
+
+		var uid = new FingerprintUid (uri, FINGERPRINT_PREFIX);
```

Comment:
We should use the disc's ID as the UID, you can write `PlaySationUid` which takes a `PlaySationHeader` as a parameter (check `SegaSaturnUid`).

Also we won't need `FINGERPRINT_PREFIX` anymore.